### ISSUE TITLE
PHPDoc for "all" helper type-hints to "string", disallowing arrays

### ISSUE THREAD

**Issue by holtkamp** [beberlei/assert#280]
The [documentation for the "all" helper states](https://github.com/beberlei/assert#all-helper):

>The Assertion::all* method checks if all provided values hold for the assertion. It will throw an exception of the assertion does not hold for one of the values:

And gives an example for an array.

Checking an array of UUIDs *works properly* but results in a warning from my IDE:

<img width="378" alt="Screenshot 2019-05-28 at 12 42 10" src="https://user-images.githubusercontent.com/776405/58472371-698c3180-8146-11e9-8f89-dc33e0dd620c.png">

This is caused by this PHPDoc:

https://github.com/beberlei/assert/blob/f812ed00763ffa96264496730b2672c744414e98/lib/Assert/Assertion.php#L108

Funny thing is, this also occurs for other "allXYZ" helpers, for example:

https://github.com/beberlei/assert/blob/f812ed00763ffa96264496730b2672c744414e98/lib/Assert/Assertion.php#L55-L57

So my question: is it correct that the PHPDoc for some "allXYZ" helpers is incorrect? I can come up with a PR when this is confirmed.

**Comment by rquadling** [comment]
Hmm. Yes. It looks like \MethodDocGenerator::generateMethodDocs needs a small bit of work to add `[]` to the first parameter typehint.

` * @method static bool allAlnum(mixed[] $value, string|callable $message = null, string $propertyPath = null) Assert that value is alphanumeric for all values.`

**Comment by rquadling** [comment]
Some work may be needed for those that are arrays (or of a type that can be thought of as an array to start with - check the allCount() method).

**Comment by holtkamp** [comment]
>Hmm. Yes. It looks like \MethodDocGenerator::generateMethodDocs needs a small bit of work

Aah, that part of the documation is generated, nice! So in that case a PR to just change the PHPDoc does not make a lot of sense...

**Comment by rquadling** [comment]
Apologies for the lack of response ... life!

```
                if ($prefix === 'all' && strpos($type, 'null') === false && $parameterIndex === 0) {
                    $type = str_replace('|', '[]|', $type).'[]';
                }
```

Seems to do the trick. 

Released in v3.2.5
